src/lf_apply.rs

```diff
@@ -750,23 +715,23 @@ pub(crate) unsafe fn rav1d_loopfilter_sbrow_rows<BD: BitDepth>(
     let endy4: c_uint = (starty4 + cmp::min((*f).h4 - sby * sbsz, sbsz)) as c_uint;
     let uv_endy4: c_uint = endy4.wrapping_add(ss_ver as c_uint) >> ss_ver;
 
-    let mut ptr: *mut BD::Pixel;
+    let mut slice: &mut [BD::Pixel];
     let mut level_ptr = &(*f).lf.level[((*f).b4_stride * sby as isize * sbsz as isize) as usize..];
-    ptr = p[0];
+    slice = p[0];
     for x in 0..(*f).sb128w {
         filter_plane_rows_y::<BD>(
             f,
             have_top,
             level_ptr,
             (*f).b4_stride,
-            ((*lflvl.offset(x as isize)).filter_y[1]).as_mut_ptr() as *const [[u16; 2]; 3],
-            ptr,
+            &(*lflvl.offset(x as isize)).filter_y[1],
+            slice.as_mut_ptr().offset(p_offset[0] as isize),
```

Comment:
If `p_offset` is a `usize` and thus positive, we can just do this, right?

```suggestion
            slice[p_offset[0]..].as_mut_ptr(),
```

Reply:
Not applicable anymore since slice is used instead of raw pointer.